docs/applications/github-action/03-azure.md

```diff
@@ -0,0 +1,128 @@
+The Massdriver CLI supports pushing images to an Azure Container Registry (ACR) and deploying them to Massdriver. This guide will walk you through the process of setting up a GitHub Action to push app changes to your ACR repository and deploy them to Massdriver.
+
+:::note
+
+Before getting started, you'll need:
+- A Massdriver organization 
+- A Massdriver [service account](/platform/service-accounts)
+- The [Massdriver CLI](https://github.com/massdriver-cloud/massdriver-cli) installed
+
+:::
+
+### Set secrets and vars
+Once you've published your application, you'll need to set the following secrets and vars in your GitHub repository:
+
+| Name | Description | Type | Notes |
+| --- | --- | --- | --- |
+| `MASSDRIVER_ORG_ID` | Your Massdriver organization ID | secret | Your Organization ID can be found by hovering over your org logo in the sidebar |
+| `MASSDRIVER_API_KEY` | Your Massdriver API key | secret | [Service Accounts](/platform/service-accounts) |
+| `AUTHENTICATION_ARTIFACT_ID_AZURE` | The ID of the authentication artifact for your Azure Container Registry | secret | Browse to your organization [artifacts](https://app.massdriver.cloud/artifacts), open your credentials artifact, and copy the UUID in the browser URL |
+| `IMAGE_REPO` | The name of your ACR repository | variable | Must be in `name/name` format. If it does not exist, it will be created for you |
+| `PACKAGE_SLUG` | The slug of your Massdriver application package | variable | Hover over the name of your application in Massdriver to view the slug |
+| `REGION` | The region where your ACR is located | variable | Must be a valid Azure region. For example: `eastus` |
+
+### Publish your application
+Before you can set up a GitHub Action to deploy your application, first you'll need to publish it to Massdriver and create a package. You can do this by following the [Getting Started](/applications/getting-started) guide.
+
+After you publish your application, you'll need to create a package. You can do this by dragging your application out from the bundle bar in the Massdriver canvas. Fill in the values of your application and click **Deploy**.
+
+### Params file
+Next, we'll need to create a params file for our application. This file will be used by the GitHub Action workflow to update the image tag and update the configuration of your application.
+
+To generate this file, in your terminal run the following command:
+
+```bash
+mass preview init <application-package-slug-id>
+```
+
+Hover over `Azure` and hit **enter** to select. Hit `s` to save. Hover over the name of your credential and hit **enter** to select. Hit `s` to save.
+
+This will output a JSON file named `preview.json` in your current directory with the values of every package in the same target as your application. Use `cat preview.json` to view the contents of the file, and copy the JSON block for your application into a new file named `mass-app-params.json`. You can find an example params file below:
```

Comment:
It’s not app Params, it’s a preview.json file. It has more in it than Params.

Reply:
This is actually a really good point, the user can grab the credential artifact UUID from here too.

I think I'll reorder these steps a bit so the customer runs `mass preview init` earlier to fetch the UUID. 

The `credentials` and `packageParams` are the only 2 categories I'm seeing in `preview.json`, is there something I'm missing?

Reply:
@coryodaniel reordered and reworded this section

Reply:
We’ll be continually adding things to this file. Secrets for example.

Reply:
Yea for sure. This is just v1 of this doc. It'll be ever evolving, I'm sure. And getting customer feedback from this doc and pain points they may experience will be good for knowing where this doc can get improvements.